src/agentscope/service/service_factory.py

```diff
@@ -50,17 +60,365 @@ def _get_type_str(cls: Any) -> Optional[Union[str, list]]:
+            if not isinstance(sub_cmd, dict):
+                raise JsonParsingError(
+                    f"Except a JSON list of dictionaries, but got"
+                    f" {type(sub_cmd)} instead.",
+                )
+
+            if "name" not in sub_cmd:
+                raise FunctionCallFormatError(
+                    "The field 'name' is required in the dictionary.",
+                )
+
+            # Obtain the service function
+            func_name = sub_cmd["name"]
+
+            # Cannot find the service function
+            if func_name not in self.service_funcs:
+                raise FunctionNotFoundError(
+                    f"Cannot find a tool function named `{func_name}`.",
+                )
+
+            # Type error for the arguments
+            if not isinstance(sub_cmd["arguments"], dict):
+                raise FunctionCallFormatError(
+                    "Except a dictionary for the arguments, but got "
+                    f"{type(sub_cmd['arguments'])} instead.",
+                )
+
+            # Leaving the type checking and required checking to the runtime
+            # error reporting during execution
+        return cmds
+
+    def _execute_func(self, cmds: List[dict]) -> str:
+        """Execute the function with the arguments.
+
+        Args:
+            cmds (`List[dict]`):
+                A list of dictionaries, where each dictionary contains the
+                name of the function and its arguments, e.g. {"name": "func1",
+                "arguments": {"arg1": 1, "arg2": 2}}.
+
+        Returns:
+            `str`: The prompt of the execution results.
+        """
+
+        execute_results = []
+        for i, cmd in enumerate(cmds):
+            func_name = cmd["name"]
+            service_func = self.service_funcs[cmd["name"]]
+            kwargs = cmd.get("arguments", {})
+
+            print(f">>> Executing function {func_name} with arguments:")
+            for key, value in kwargs.items():
+                value = (
+                    value if len(str(value)) < 50 else str(value)[:50] + "..."
+                )
+                print(f">>> \t{key}: {value}")
+
+            # Execute the function
+            try:
+                func_res = service_func.processed_func(**kwargs)
+            except Exception as e:
+                func_res = ServiceResponse(
+                    status=ServiceExecStatus.ERROR,
+                    content=str(e),
+                )
+
+            print(">>> END ")
+
+            status = (
+                "SUCCESS"
+                if func_res.status == ServiceExecStatus.SUCCESS
+                else "FAILED"
+            )
+
+            arguments = [f"{k}: {v}" for k, v in kwargs.items()]
+
+            execute_res = self.execute_result_template.format_map(
+                {
+                    "index": i + 1,
+                    "function_name": cmd["name"],
+                    "arguments": "\n\t\t".join(arguments),
```

Comment:
why use `\t\t` here

Reply:
Because the arguments are described in prompt as follows, which requires "/t/t"
```python
execute_result_template = (
        "{index}. Execute function {function_name}\n"
        "   [EXECUTE ARGUMENTS]:\n"
        "       {arguments}\n"
        "   [EXECUTE STATUS]: {status}\n"
        "   [EXECUTE RESULT]: {result}\n"
    )
```